sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsPartitionManagement.java

```diff
@@ -139,4 +139,21 @@ void replacePartitionMetadata(
      * @return an array of Identifiers for the partitions
      */
     InternalRow[] listPartitionIdentifiers(String[] names, InternalRow ident);
+
+    /**
+     * Rename an existing partition of the table.
+     *
+     * @param from an existing partition identifier to rename
+     * @param to new partition identifier
+     * @return true if renaming completes successfully otherwise false
+     * @throws UnsupportedOperationException If partition renaming is not supported
+     * @throws PartitionAlreadyExistsException If the `to` partition exists already
+     * @throws NoSuchPartitionException If the `from` partition does not exist
+     */
```

Comment:
Ah I missed that we should add `@since 3.2.0`. @MaxGekk please add it in a followup.

Reply:
Here it is: https://github.com/apache/spark/pull/30964